extensions/platform-http/runtime/src/main/java/org/apache/camel/quarkus/component/platform/http/runtime/PlatformHttpRecorder.java

```diff
@@ -0,0 +1,24 @@
+package org.apache.camel.quarkus.component.platform.http.runtime;
+
+import org.apache.camel.component.platform.http.PlatformHttpComponent;
+import org.apache.camel.component.platform.http.PlatformHttpConstants;
+import org.apache.camel.component.platform.http.spi.PlatformHttpEngine;
+import org.apache.camel.quarkus.core.runtime.CamelRuntime;
+import org.apache.camel.spi.Registry;
+
+import io.quarkus.arc.runtime.BeanContainer;
+import io.quarkus.runtime.RuntimeValue;
+import io.quarkus.runtime.annotations.Recorder;
+
+@Recorder
+public class PlatformHttpRecorder {
+
+    public void registerBeans(RuntimeValue<CamelRuntime> runtime, BeanContainer beanContainer) {
+        final Registry registry = runtime.getValue().getRegistry();
+        final QuarkusPlatformHttpEngine engine = beanContainer.instance(QuarkusPlatformHttpEngine.class);
```

Comment:
Can't the `QuarkusPlatformHttpEngine` be a parameter of the recorder so that the bean container is not involved ?

Reply:
`QuarkusPlatformHttpEngine` needs to `@Inject` the `Router`. How could that work without the bean container?

Reply:
There a router build item

Reply:
Indeed, but only since Quarkus 0.23. I'd prefer improving this later when we are on 0.23

Reply:
In bb474bb, the bean container is not involved. The `QuarkusPlatformHttpEngine` is created in the recorder though, which is in line with the requirement of Quarkus that the runtime objects should not be created inside BuildSteps.